Adds VideoEncoder (#8989)
mdn/content: files/en-us/web/api/videoencoder/configure/index.md

---
title: VideoEncoder.configure()
slug: Web/API/VideoEncoder/configure
tags:
  - API
  - Method
  - Reference
  - configure
  - VideoEncoder
browser-compat: api.VideoEncoder.configure
---
{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}

> The **`configure`** method of the {{domxref("VideoEncoder")}} interface enqueues a control message to configure the audio decoder for decoding chunks.

Reviewer: ```suggestion
The **`configure()`** method of the {{domxref("VideoEncoder")}} interface enqueues a control message to configure the audio decoder for decoding chunks.
```

## Syntax

```js
AudioDecoder.configure(config)
```

### Parameters

- `config`
  - : A dictionary object containing the following members:
    - `codec`
      - : A {{domxref("DOMString","string")}} containing a [valid codec string](https://www.w3.org/TR/webcodecs-codec-registry/#audio-codec-registry).
    - `width`{{Optional_Inline}}
      - : An integer representing the width of each output {{domxref("EncodedVideoChunk")}} in pixels, before any ratio adjustments.
    - `height`{{Optional_Inline}}
      - : An integer representing the height of each output {{domxref("EncodedVideoChunk")}} in pixels, before any ratio adjustments.
    - `displayWidth`{{Optional_Inline}}
      - : An integer representing the intended display width of each output {{domxref("EncodedVideoChunk")}} in pixels when displayed.
    - `displayHeight`{{Optional_Inline}}
      - : An integer representing the vertical dimension of each output {{domxref("EncodedVideoChunk")}} in pixels when displayed.
    - `hardwareAcceleration`

>       - : An hint that configures the hardware accelaration method of this codec. One of:

Reviewer: ```suggestion
      - : A hint that configures the hardware acceleration method of this codec. One of:
```

        - `"no-preference"`
        - `"prefer-hardware"`
        - `"prefer-software"`
    - `bitrate`
      - : An integer containing the average bitrate of the encoded video in units of bits per second.
    - `framerate`

>       - : An integer containing the expected framerate in frames per second.

Reviewer: ```suggestion
      - : An integer containing the expected frame rate in frames per second.
```

    - `alpha`

>       - : A string indicating whether the alpha component of the VideoFrame inputs should be kept or discarded prior to encoding, defaults to `"discard"`. One of:

Reviewer: ```suggestion
      - : A string indicating whether the alpha component of the `VideoFrame` inputs should be kept or discarded prior to encoding. One of:
```

Reviewer: This goes with the next one. This optimizes for scanning.

>         - `"discard"`

Reviewer: ```suggestion
        - `"discard"` (default)
```

        - `"keep"`
    - `scalabilityMode`
      - : A {{domxref("DOMString", "string")}} containing an encoding scalability mode identifier as defined in [WebRTC](https://w3c.github.io/webrtc-svc/#scalabilitymodes*).
    - `bitrateMode`

>       - : A string containing a bitrate mode, defaulting to `"variable"`. One of:

Reviewer: ```suggestion
      - : A string containing a bitrate mode. One of:
```

        - `"constant"`

>         - `"variable"`

Reviewer: ```suggestion
        - `"variable"` (default)
```

    - `latencyMode`

>       - : A string containing a value that configures the latency behavior of this codec, defaulting to `"quality"`. One of:

Reviewer: ```suggestion
      - : A string containing a value that configures the latency behavior of this codec. One of:
```

>         - `"quality"`

Reviewer: ```suggestion
        - `"quality"` (default)
```
